### PR TITLE
feat: implement snowflake algorithm

### PR DESCRIPTION
**Pull request by azhsmesos**
解决两个问题：
 1.时钟回拨，没有采用redis的方式，而是本地保存当前最大timestart，每次启动时就找到这台机器上面的max_time自增就好了，保持逻辑时钟
 2.跨毫秒起始值起始每次默认都是0，这会导致末尾是偶数不方便取余，所以采用random的方式选取1～3的随机数

### CODE REVIEW

**Review by seeflood — CHANGES_REQUESTED** [review]
need to implement GetSegment

**Comment by seeflood** [comment]
action:
调研一下要不要接存储
调研一下时钟回拨问题

File: components/sequencer/snowflake/sequencer_snowflake.go
Comment (seeflood) [inline components/sequencer/snowflake/sequencer_snowflake.go:26]: Need english comments

**Review by seeflood — CHANGES_REQUESTED** [review]

**Comment by ZLBer** [comment]
@seeflood 
This subject participated in GLCC.I think we should follow the rules of GLCC.
PR should not be submitted now.It's not fair.
We should choose the candidate from all students who have submitted their proposal.

**Comment by seeflood** [comment]
@ZLBer Agreed. We should follow the rules and make it fair.
@azhsmesos Thanks for your contribution, but please submit a GLCC proposal and wait for the approval before submitting this PR.
We have to close this PR and wait for the GLCC process.
For now, you can continue with other PR, like #554 and #137

**Pull request closed without merging**
